Tasks/DownloadFileShareArtifactsV0/main.ts

```diff
@@ -0,0 +1,105 @@
+
+var taskJson = require('./task.json');
+
+tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+const area: string = 'DownloadFileShareArtifacts';
+
+function getDefaultProps() {
+    var hostType = (tl.getVariable('SYSTEM.HOSTTYPE') || "").toLowerCase();
+    return {
+        hostType: hostType,
+        definitionName: '[NonEmail:' + (hostType === 'release' ? tl.getVariable('RELEASE.DEFINITIONNAME') : tl.getVariable('BUILD.DEFINITIONNAME')) + ']',
+        processId: hostType === 'release' ? tl.getVariable('RELEASE.RELEASEID') : tl.getVariable('BUILD.BUILDID'),
+        processUrl: hostType === 'release' ? tl.getVariable('RELEASE.RELEASEWEBURL') : (tl.getVariable('SYSTEM.TEAMFOUNDATIONSERVERURI') + tl.getVariable('SYSTEM.TEAMPROJECT') + '/_build?buildId=' + tl.getVariable('BUILD.BUILDID')),
+        taskDisplayName: tl.getVariable('TASK.DISPLAYNAME'),
+        jobid: tl.getVariable('SYSTEM.JOBID'),
+        agentVersion: tl.getVariable('AGENT.VERSION'),
+        agentOS: tl.getVariable('AGENT.OS'),
+        agentName: tl.getVariable('AGENT.NAME'),
+        version: taskJson.version
+    };
+}
+
+function publishEvent(feature, properties: any): void {
+    try {
+        var splitVersion = (process.env.AGENT_VERSION || '').split('.');
+        var major = parseInt(splitVersion[0] || '0');
+        var minor = parseInt(splitVersion[1] || '0');
+        let telemetry = '';
+        if (major > 2 || (major == 2 && minor >= 120)) {
+            telemetry = `##vso[telemetry.publish area=${area};feature=${feature}]${JSON.stringify(Object.assign(getDefaultProps(), properties))}`;
+        }
+        else {
+            if (feature === 'reliability') {
+                let reliabilityData = properties;
+                telemetry = "##vso[task.logissue type=error;code=" + reliabilityData.issueType + ";agentVersion=" + tl.getVariable('Agent.Version') + ";taskId=" + area + "-" + JSON.stringify(taskJson.version) + ";]" + reliabilityData.errorMessage
+            }
+        }
+        console.log(telemetry);;
+    }
+    catch (err) {
+        tl.warning("Failed to log telemetry, error: " + err);
+    }
+}
+
+async function main(): Promise<void> {
+    var promise = new Promise<void>(async (resolve, reject) => {
+        var downloadPath: string = tl.getInput("downloadPath", true);
+        var debugMode: string = tl.getVariable('System.Debug');
+        var isVerbose: boolean = debugMode ? debugMode.toLowerCase() != 'false' : false;
+        var parallelLimit: number = +tl.getInput("parallelizationLimit", false);
+        var retryLimit = parseInt(tl.getVariable("VSTS_HTTP_RETRY")) ? parseInt(tl.getVariable("VSTS_HTTP_RETRY")) : 4;
+        var itemPattern: string = tl.getInput("itemPattern", false) || '**';
+
+        let downloader = new engine.ArtifactEngine();
+        let downloadUrl = tl.getInput("location", true);
+        let artifactName = tl.getInput("artifactName", true);
```

Comment:
do we need to replace / with \ as done here?
https://github.com/Microsoft/vsts-tasks/blob/fb86546adc1dfeca3602a2887891f893dc732823/Tasks/DownloadBuildArtifactsV0/main.ts#L251